crates/bevy_reflect/src/reflect.rs

```diff
@@ -42,15 +42,15 @@ pub enum ReflectMut<'a> {
     Array(&'a mut dyn Array),
     Map(&'a mut dyn Map),
     Enum(&'a mut dyn Enum),
-    Value(&'a mut dyn Reflect),
+    Value(&'a mut dyn PartialReflect),
```

Comment:
the problem with this is trying to support proxies for scripting types in the future, which in my understanding should implement `PartialReflect` but not `Reflect`. given a javascript object like `{ a: number, b: number }` it's easy to see that it can implement `reflect_ref` to return `ReflectRef::Struct` since it has fields. however, it's not so clear cut with the primitive `number`: should it be a `Value` of some type like `JsNumber`?

`ReflectRef::Value` in my understanding is for Rust types only.

> Since all proxy types are structural.

this is only true for the proxy types *we* have currently. other crates may reasonably want to have a non-structural proxy.

i agree that `ReflectRef::Value` should always implement `Reflect` since `ValueInfo` has [an associated `TypeId`](https://docs.rs/bevy/latest/bevy/reflect/struct.ValueInfo.html#method.type_id) but if we do this i think we in the future will need a new variant (e.g. `ReflectRef::Opaque(dyn PartialReflect)` and a unit `TypeInfo::Opaque` variant) because this will break conceptually non-Rust types.

currently a `ReflectRef::Opaque` would be effectively useless since it could neither be downcasted or referenced in the registry so i'm happy to make the change here but once the state of reflection scripting improves we might want to add `Opaque` or revert this change.